### PR TITLE
proper session restoration

### PR DESCRIPTION
**Pull request by zwhitchcox**
Currently, if you open a session after vim is already opened (not with `vim -S ./path/to/session.vim`), the old `nvim-tree` buffers are not restored. This ensures that `nvim-tree` is reloaded whenever a session is loaded.

### CODE REVIEW

**Comment by zwhitchcox** [comment]
So, I know this code is wrong, but I'm not sure how, so I was hoping I could get some feedback on it 😅

**Comment by zwhitchcox** [comment]
cc @kyazdani42  (from https://github.com/nvim-tree/nvim-tree.lua/issues/591#issuecomment-912728592)

**Review by kyazdani42 — COMMENTED** [review]
looks good to me. cc @alex-courtis

**Comment by zwhitchcox** [comment]
Just made another commit...I was getting an error to save the `NvimTree_*` file before exiting before this fix.

File: lua/nvim-tree/view.lua [hunk 1]
```diff
@@ -74,16 +74,24 @@ local function matches_bufnr(bufnr)
...
-local function wipe_rogue_buffer()
+function M.wipe_rogue_buffer()
+  local was_wiped = false
   for _, bufnr in ipairs(vim.api.nvim_list_bufs()) do
     if not matches_bufnr(bufnr) and utils.is_nvim_tree_buf(bufnr) then
+
+      -- if restored from session, values might be incorrect
+      for option, value in pairs(BUFFER_OPTIONS) do
+        vim.bo[bufnr][option] = value
+      end
```
Comment (kyazdani42) [inline lua/nvim-tree/view.lua:85]: this is outside the scope of this function though.
Also if the buffer is deleted, i'm not sure what's the point of resetting its options.

Comment (zwhitchcox) [reply lua/nvim-tree/view.lua:85]: I was having to do this due to a misconfiguration in my setup. I'm having another issue now. Going to close this PR for now until I fix it.
